process/coordinator/process.go

```diff
@@ -113,17 +113,18 @@ func NewTransactionCoordinator(
 }
 
 // separateBodyByType creates a map of bodies according to type
-func (tc *transactionCoordinator) separateBodyByType(body block.Body) map[block.Type]block.Body {
-	separatedBodies := make(map[block.Type]block.Body)
-
-	for i := 0; i < len(body); i++ {
-		mb := body[i]
+func (tc *transactionCoordinator) separateBodyByType(body *block.Body) map[block.Type]*block.Body {
+	separatedBodies := make(map[block.Type]*block.Body)
+	if !check.IfNil(body) {
```

Comment:
please use this construction instead

if check.IfNil(body){
return ...
}

...

Reply:
done